packages/teleterm/src/services/pty/resolveShellEnv.ts

```diff
@@ -0,0 +1,139 @@
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/**
+ MIT License
+
+ Copyright (c) 2015 - present Microsoft Corporation
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+//Based on https://github.com/microsoft/vscode/blob/main/src/vs/platform/shell/node/shellEnv.ts
+
+import { Logger } from 'shared/libs/logger';
+import { unique } from 'teleterm/ui/utils/uid';
+import { spawn } from 'child_process';
+import { memoize } from 'lodash';
+
+const logger = new Logger('resolveShellEnv()');
+const resolveShellMaxTime = 8000; // 8s
+
+export const resolveShellEnvCached = memoize(resolveShellEnv);
+
+async function resolveShellEnv(shell: string): Promise<typeof process.env> {
+  if (process.platform === 'win32') {
+    logger.trace('skipped Windows platform');
+    return;
+  }
+  // TODO(grzegorz) skip if already running from CLI
+
+  const timeoutController = new AbortController();
+  const timeout = setTimeout(() => {
+    timeoutController.abort();
+  }, resolveShellMaxTime);
+
+  try {
+    return await resolveUnixShellEnv(shell, timeoutController.signal);
+  } finally {
+    clearTimeout(timeout);
+  }
+}
+
+async function resolveUnixShellEnv(
+  shell: string,
+  abortSignal: AbortSignal,
+): Promise<typeof process.env> {
+  const mark = unique().replace(/-/g, '').substring(0, 12);
+  const regex = new RegExp(mark + '(.*)' + mark);
+
+  const env = {
+    ...process.env,
+    ELECTRON_RUN_AS_NODE: '1',
+    ELECTRON_NO_ATTACH_CONSOLE: '1',
+  };
+
+  return new Promise<typeof process.env>((resolve, reject) => {
+    const command = `'${process.execPath}' -p '"${mark}" + JSON.stringify(process.env) + "${mark}"'`;
+    const shellArgs = shell === '/bin/tcsh' ? ['-ic'] : ['-ilc'];
```

Comment:
Does it have to be `interactive`?

Reply:
Matter of fact, we should perhaps link to this stackexchange post in case anyone has the same question in the future. ;)